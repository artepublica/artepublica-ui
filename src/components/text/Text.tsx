import { Text as NativeText, StyleSheet, TextStyle } from 'react-native';

import { Theme, useTheme } from '@utils';

type TextProps = {
    testID?: string;
    children?: string | number | (string | number)[] | null;
    numberOfLines?: number;
    style?: TextStyle | TextStyle[];
};

function Text({ testID, children, numberOfLines, style }: TextProps): JSX.Element {
    const { theme } = useTheme();

    const textStyle = styles(theme);

    return (
        <NativeText
            testID={testID}
            numberOfLines={numberOfLines}
            style={[
                textStyle.text,
                style,
            ]}
        >
            {children}
        </NativeText>
    );
}

export default Text;

function styles(theme: Theme) {
    return StyleSheet.create({
        text: {
            fontFamily: theme.text.font.fontFamily,
            color: theme.text.textColor,
        },
    });
}
