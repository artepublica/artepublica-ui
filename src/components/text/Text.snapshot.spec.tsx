import { create } from 'react-test-renderer';

import Text from './Text';

describe('<Text />', () => {
    it('default snapshot', () => {
        const label = 'test';
        const json = create(<Text>{label}</Text>).toJSON();

        expect(json).toMatchSnapshot();
    });
});
