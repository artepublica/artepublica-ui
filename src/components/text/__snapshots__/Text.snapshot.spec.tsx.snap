// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> default snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "white",
        "fontFamily": "fontFamily",
      },
      undefined,
    ]
  }
>
  test
</Text>
`;
